puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/source/katello.groovy

```diff
@@ -0,0 +1,157 @@
+def commit_hash = ''
+foreman_branch = 'master'
+project_name = 'katello'
+source_type = 'gem'
+ruby = '2.5'
+
+pipeline {
+    options {
+        timestamps()
+        timeout(time: 2, unit: 'HOURS')
+        ansiColor('xterm')
+    }
+
+    agent { label 'fast' }
+
+    stages {
+        stage('Setup Git Repos') {
+            steps {
+                deleteDir()
+                git url: "https://github.com/katello/${project_name}", branch: foreman_branch, poll: false
+
+                dir('foreman') {
+                   git url: "https://github.com/theforeman/foreman", branch: 'develop', poll: false
+                }
+            }
+        }
+        stage("Setup RVM") {
+            steps {
+
+                configureRVM(ruby)
+
+            }
+        }
+        stage('Configure Environment') {
+            steps {
+
+                dir('foreman') {
+                    addGem()
+                    databaseFile(gemset())
+                }
+
+            }
+        }
+        stage('Configure Database') {
+            steps {
+
+                dir('foreman') {
+                    configureDatabase(ruby)
+                }
+
+            }
+        }
+        stage('Run Tests') {
+            parallel {
+                stage('tests') {
+                    steps {
+                        dir('foreman') {
+                            withRVM(['bundle exec rake jenkins:katello TESTOPTS="-v" --trace'], ruby)
+                        }
+                    }
+                }
+                stage('rubocop') {
+                    steps {
+                        dir('foreman') {
+                            withRVM(['bundle exec rake katello:rubocop TESTOPTS="-v" --trace'], ruby)
+                        }
+                    }
+                }
+                stage('react-ui') {
+                    when {
+                        expression { fileExists('package.json') }
+                    }
+                    steps {
+                        sh "npm install"
+                        sh 'npm run lint'
+                        sh 'npm test'
+                    }
+                }
+                stage('angular-ui') {
```

Comment:
this stage will only work for `master`, do you intend to switch other katello branches to this model too?

Reply:
This job is all about releasing and building katello source for master only.